logging/logging-processor/src/main/java/org/infinispan/logging/processor/report/XmlReportWriter.java

```diff
@@ -0,0 +1,93 @@
+package org.infinispan.logging.processor.report;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+import javax.lang.model.element.Element;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import org.infinispan.logging.annotations.Description;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+/**
+ * XmlReportWriter.
+ * @author Durgesh Anaokar
+ * @Since 13.0
+ */
+public class XmlReportWriter implements AutoCloseable {
```

Comment:
Most of the methods in this class have `throws IOException`, however what they really do is throw `XMLStreamException` and simply wrap this as `IOException`. We can remove the try/catch from all of the methods if we simply declare `throws XMLStreamException` on the methods and then catch `XMLStreamException` instances in `InfinispanLoggingProcessor#createReportXml`

The only exception is for `close()` which must throw IOException as part of the `AutoCloseable` contract.